elementary/monitor/data_monitoring/data_monitoring_report.py

```diff
@@ -0,0 +1,343 @@
+import json
+import os
+import os.path
+import re
+import webbrowser
+from collections import defaultdict
+from typing import Dict, List, Optional, Tuple
+
+import pkg_resources
+
+from elementary.clients.gcs.client import GCSClient
+from elementary.clients.s3.client import S3Client
+from elementary.config.config import Config
+from elementary.monitor.api.filters.filters import FiltersAPI
+from elementary.monitor.api.lineage.lineage import LineageAPI
+from elementary.monitor.api.lineage.schema import LineageSchema
+from elementary.monitor.api.models.models import ModelsAPI
+from elementary.monitor.api.models.schema import (
+    ModelRunsSchema,
+    NormalizedExposureSchema,
+    NormalizedModelSchema,
+    NormalizedSourceSchema,
+)
+from elementary.monitor.api.sidebar.schema import SidebarsSchema
+from elementary.monitor.api.sidebar.sidebar import SidebarAPI
+from elementary.monitor.api.tests.schema import TotalsSchema
+from elementary.monitor.api.tests.tests import TestsAPI
+from elementary.monitor.data_monitoring.data_monitoring import DataMonitoring
+from elementary.monitor.data_monitoring.schema import DataMonitoringFilter
+from elementary.tracking.anonymous_tracking import AnonymousTracking
+from elementary.utils.log import get_logger
+from elementary.utils.time import get_now_utc_iso_format
+
+logger = get_logger(__name__)
+
+YAML_FILE_EXTENSION = ".yml"
+SQL_FILE_EXTENSION = ".sql"
+
+
+class DataMonitoringReport(DataMonitoring):
+    def __init__(
+        self,
+        config: Config,
+        tracking: AnonymousTracking,
+        force_update_dbt_package: bool = False,
+        disable_samples: bool = False,
+        filter: Optional[str] = None,
+    ):
+        super().__init__(config, tracking, force_update_dbt_package, disable_samples)
+        self.filter = self._parse_filter(filter)
+        self.tests_api = TestsAPI(dbt_runner=self.internal_dbt_runner)
+        self.models_api = ModelsAPI(dbt_runner=self.internal_dbt_runner)
+        self.sidebar_api = SidebarAPI(dbt_runner=self.internal_dbt_runner)
+        self.lineage_api = LineageAPI(dbt_runner=self.internal_dbt_runner)
+        self.filter_api = FiltersAPI(dbt_runner=self.internal_dbt_runner)
+        self.s3_client = S3Client.create_client(self.config, tracking=self.tracking)
+        self.gcs_client = GCSClient.create_client(self.config, tracking=self.tracking)
+
+    def _parse_filter(self, filter: Optional[str] = None) -> DataMonitoringFilter:
+        data_monitoring_filter = DataMonitoringFilter()
+        if filter:
+            invocation_id_regex = re.compile(r"invocation_id:\w+")
+            invocation_time_regex = re.compile(
+                r"invocation_time:([\+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00))?(\17[0-5]\d)))"
+            )
```

Comment:
Do we actually need the full huge regex?
Can we maybe only rely on the "invocation_id:" or "invocation_time:" prefix instead?

I think it makes sense to first parse the prefix, and then if needed show an error that's specific to the prefix used (e.g. "Invalid timestamp provided for invocation_time selector", which you can get from regular datetime parsing)